lib/project_types/script/layers/infrastructure/script_service.rb

```diff
@@ -34,11 +34,13 @@ def set_app_script(
             configurationUi: script_config.configuration_ui,
             configurationDefinition: script_config.configuration&.to_json,
             moduleUploadUrl: module_upload_url,
-            library: {
-              language: library[:language],
-              version: library[:version],
-            },
           }
+
+          variables.merge!({
+            language: library[:language],
+            version: library[:version],
+          }) if library
```

Comment:
nit: Its not a big deal in this situation since the hash is small and won't be noticeable to any human waiting, but its more efficient to set the hash keys instead of merging two hashes
```suggestion
          if library
            variables[:language] = library[:language]
            variables[:version] = library[:version]
          end
```

Also - is there any use in setting the language to wasm? or does it not matter since those variables are just for library stuff?

Reply:
Good to know! I think we should leave the language undefined in this case, because in this context we're referring to the library_language, and there's no associated library.